core/api/views/projects.py

```diff
@@ -73,6 +78,53 @@ class MetaProjectListView(generics.ListAPIView):
     serializer_class = MetaProjectSerializer
 
 
+class MetaProjectMyaListView(generics.ListAPIView):
+    """
+    List meta projects available for MYA update.
+    """
+
+    permission_classes = [HasMetaProjectsViewAccess]
+    serializer_class = MetaProjecMyaSerializer
+    filterset_class = MetaProjectMyaFilter
+
+    def get_queryset(self):
+        result = (
+            MetaProject.objects.filter(
+                type=MetaProject.MetaProjectType.MYA,
+                projects__submission_status__name="Approved",
+            )
+            .exclude(
+                projects__status__name="Completed",
```

Comment:
The exclude in Django will remove from the query all meta projects that have at least one project with the status "Completed". Not sure if this was the desired behaviour.

If you want to exclude only metaprojects which have all their projects "Completed", try something like:

```
MetaProject.objects.filter(
    type=MetaProject.MetaProjectType.MYA,
    projects__submission_status__name="Approved",
).filter(
    Exists(
        Project.objects.filter(
            metaproject=OuterRef('pk')
        ).exclude(
            status__name="Completed"
        )
    )
)